crates/router/src/core/admin.rs

```diff
@@ -106,6 +106,69 @@ fn add_publishable_key_to_decision_service(
     );
 }
 
+#[cfg(feature = "olap")]
+pub async fn create_organization(
+    state: SessionState,
+    req: api::OrganizationRequest,
+) -> RouterResponse<api::OrganizationResponse> {
+    let db_organization = ForeignFrom::foreign_from(req);
+    state
+        .store
+        .insert_organization(db_organization)
+        .await
+        .to_duplicate_response(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Error when creating organization")
+        .map(ForeignFrom::foreign_from)
+        .map(service_api::ApplicationResponse::Json)
+}
+
+#[cfg(feature = "olap")]
+pub async fn update_organization(
+    state: SessionState,
+    org_id: api::OrganizationId,
+    req: api::OrganizationRequest,
+) -> RouterResponse<api::OrganizationResponse> {
+    let organization_update = diesel_models::organization::OrganizationUpdate::Update {
+        org_name: req.organization_name,
+    };
+    state
+        .store
+        .update_organization_by_org_id(&org_id.organization_id, organization_update)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::GenericNotFoundError {
+            message: "organization with the given id does not exist".to_string(),
+        })
+        .attach_printable(format!(
+            "Failed to update organization with organization_id: {}",
+            org_id.organization_id
+        ))
+        .map(ForeignFrom::foreign_from)
+        .map(service_api::ApplicationResponse::Json)
+}
+
+#[cfg(feature = "olap")]
+pub async fn get_organization(
+    state: SessionState,
+    org_id: api::OrganizationId,
+) -> RouterResponse<api::OrganizationResponse> {
+    #[cfg(all(not(feature = "v2"), feature = "olap"))]
```

Comment:
use v1 and v2 conventions

Reply:
This change will be made once this pr gets merged.
https://github.com/juspay/hyperswitch/pull/5365